src/SingleProject/Resizetizer/test/UnitTests/BaseTest.cs

```diff
@@ -0,0 +1,179 @@
+			}
+			else
+			{
+				var name = GetType().FullName;
+				if (name.StartsWith(TestFolderName + ".", StringComparison.OrdinalIgnoreCase))
+					name = name.Substring(TestFolderName.Length + 1);
+
+				var dir = Path.Combine(Path.GetTempPath(), TestFolderName, name, Path.GetRandomFileName());
+
+				DestinationDirectory = dir;
+				DeleteDirectory = dir;
+			}
+		}
+
+		public virtual void Dispose()
+		{
+			if (Directory.Exists(DeleteDirectory))
+				Directory.Delete(DeleteDirectory, true);
+		}
+
+		protected void AssertFileSize(string file, int width, int height)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			Assert.True(File.Exists(file), $"File did not exist: {file}");
+
+			using var codec = SKCodec.Create(file);
+			Assert.Equal(new SKSizeI(width, height), codec.Info.Size);
+		}
+
+		protected void AssertFileExists(string file)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			Assert.True(File.Exists(file), $"File did not exist: {file}");
+		}
+
+		protected void AssertFileNotExists(string file)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			Assert.False(File.Exists(file), $"File existed: {file}");
+		}
+
+		protected void AssertFileContains(string file, params string[] snippet)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			var content = File.ReadAllText(file);
+
+			foreach (var snip in snippet)
+				Assert.Contains(snip, content, StringComparison.Ordinal);
+		}
+
+		protected void AssertFileContains(string file, SKColor color, int x, int y)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			using var resultImage = SKBitmap.Decode(file);
+			Assert.Equal(color, resultImage.GetPixel(x, y));
+		}
+
+		protected void AssertFileDoesNotContain(string file, SKColor color, int x, int y)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			using var resultImage = SKBitmap.Decode(file);
+			Assert.NotEqual(color, resultImage.GetPixel(x, y));
+		}
+
+		protected void AssertFileContains(string file, params SKColor[] colors)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			using var resultImage = SKBitmap.Decode(file);
+			var pixels = resultImage.Pixels;
+			foreach (var color in colors)
+				Assert.Contains(color, pixels);
+		}
+
+		protected void AssertFileDoesNotContain(string file, params SKColor[] colors)
+		{
+			file = Path.Combine(DestinationDirectory, file);
+
+			using var resultImage = SKBitmap.Decode(file);
+			var pixels = resultImage.Pixels;
+			foreach (var color in colors)
+				Assert.DoesNotContain(color, pixels);
+		}
+
+		protected void AssertFileMatches(string destinationFilename, object[] args = null, [CallerMemberName] string methodName = null) =>
+			AssertFileMatchesReal(destinationFilename, args, methodName);
+
+		protected void SaveImageResultFile(string destinationFilename, object[] args = null, [CallerMemberName] string methodName = null) =>
+			SaveImageResultFileReal(destinationFilename, args, methodName);
```

Comment:
These 2 methods can be used to generate (in source folder) and then test/compare results. Typically, you would:

1. use `SaveImageResultFile` locally at the end of the tests with all the files you want
2. ensure the image is correct
3. replace with `AssertFileMatches` 
4. push to git

CI will come along and run the tests and compare. And, if you use a smart git tool, the tool can show you diffs of git files and you can use runs locally to ensure you did not break anything.

Reply:
Did you check if we could use an assertion library like Verify?

https://github.com/VerifyTests/Verify

Might simplify some of the test code.

There is an older one called `ApprovalTests`, but it only worked on Windows.